roachtest/pg_regress: accept recent diffs

yuzefovich opened the pull request:
This commit accepts the recent diffs. Several issues have been filed for the differences observed (some are actually already fixed, plus this commit assumes that #145246 is fixed too - it currently has a fix in review).

Most notable changes in the diff:
- jsonpath improvements,
- RLS changes,
- fix to $1 notation of PLpgSQL as well as support of set-returning PLpgSQL functions.

Additionally, this commit improves the runner a bit to remove table IDs from error messages to reduce diff churn.

Fixes: #144338
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/145233)
<!-- Reviewable:end -->

yuzefovich committed:
roachtest/pg_regress: accept recent diffs

This commit accepts the recent diffs. Several issues have been filed for
the differences observed (some are actually already fixed, plus this
commit assumes that 145246 is fixed too - it currently has a fix in
review).

Most notable changes in the diff:
- jsonpath improvements,
- RLS changes,
- fix to $1 notation of PLpgSQL as well as support of set-returning
PLpgSQL functions.

Additionally, this commit improves the runner a bit to remove table IDs
from error messages to reduce diff churn.

Release note: None

DrewKimball reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> Thanks! I commented on the places I noticed that seem wrong. Happy to open issues as well if you haven't gotten to them yet.

> Reviewed 36 of 36 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/145233#-:-OP-J80k3tUMomOH9bgB:b-98idjr)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @yuzefovich)

___
*[`pkg/cmd/roachtest/testdata/pg_regress/jsonpath.diffs` line 140 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/145233#-OP-B4IS6cLZAX2aBWb5:-OP-B4IS6cLZAX2aBWb6:bf96zrx) ([raw file](https://github.com/cockroachdb/cockroach/blob/6f780acb813cce431d78b6ee7b331e189b75bad9/pkg/cmd/roachtest/testdata/pg_regress/jsonpath.diffs#L140)):*
> ```
> +  jsonpath  
> +------------
> + ($"" - -1)
> ```

Looks like something strange going on with the normalization, here.
___
*[`pkg/cmd/roachtest/testdata/pg_regress/jsonpath.diffs` line 218 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/145233#-OP-BaTq1dE0Q0j3jCe3:-OP-BaTq1dE0Q0j3jCe4:b1fk0o2) ([raw file](https://github.com/cockroachdb/cockroach/blob/6f780acb813cce431d78b6ee7b331e189b75bad9/pkg/cmd/roachtest/testdata/pg_regress/jsonpath.diffs#L218)):*
> ```
> + jsonpath 
> +----------
> + 1.type()
> ```

This is another case where the normalization isn't quite right - the result doesn't round-trip through the jsonpath parser:
```
root@localhost:26257/defaultdb> select '1.type()'::jsonpath;
ERROR: could not parse "1.type()" as type jsonpath: at or near "trailing junk after numeric literal at or near "1.t"": syntax error
SQLSTATE: 42601
DETAIL: source SQL:
1.type()
^
```
___
*[`pkg/cmd/roachtest/testdata/pg_regress/jsonpath.diffs` line 227 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/145233#-OP-BzoA2Nc15Yt2IrsE:-OP-BzoA2Nc15Yt2IrsF:bhycl3c) ([raw file](https://github.com/cockroachdb/cockroach/blob/6f780acb813cce431d78b6ee7b331e189b75bad9/pkg/cmd/roachtest/testdata/pg_regress/jsonpath.diffs#L227)):*
> ```
> +  jsonpath  
> +------------
> + 1.2.type()
> ```

Interestingly, this case does parse, unlike the one above.
___
*[`pkg/cmd/roachtest/testdata/pg_regress/jsonpath.diffs` line 358 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/145233#-OP-CLqx4ZlXtObbsmlJ:-OP-CLqx4ZlXtObbsmlK:bxwtizn) ([raw file](https://github.com/cockroachdb/cockroach/blob/6f780acb813cce431d78b6ee7b331e189b75bad9/pkg/cmd/roachtest/testdata/pg_regress/jsonpath.diffs#L358)):*
> ```
> +                    jsonpath                     
> +-------------------------------------------------
> + (($ + 1)."a" + 2."b"?(@ > 1 || exists (@."c")))
> ```

This is another one that doesn't round-trip.
___
*[`pkg/cmd/roachtest/testdata/pg_regress/plpgsql.diffs` line 4100 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/145233#-OP-F1tY7RUkWXXxIItU:-OP-F1tY7RUkWXXxIItV:bnj5xe0) ([raw file](https://github.com/cockroachdb/cockroach/blob/6f780acb813cce431d78b6ee7b331e189b75bad9/pkg/cmd/roachtest/testdata/pg_regress/plpgsql.diffs#L4100)):*
> ```
> +   |  
> +   |  
>  (5 rows)
> ```

I opened the issue for this one here: https://github.com/cockroachdb/cockroach/issues/145407
___
*[`pkg/cmd/roachtest/testdata/pg_regress/temp.diffs` line 471 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/145233#-OP-Hqvw6SYZS84rstvu:-OP-Hqvw6SYZS84rstvv:b-39b4to) ([raw file](https://github.com/cockroachdb/cockroach/blob/6f780acb813cce431d78b6ee7b331e189b75bad9/pkg/cmd/roachtest/testdata/pg_regress/temp.diffs#L471)):*
> ```
>  create view pg_temp.twophase_view as select 1;
> +NOTICE:  auto-committing transaction before processing DDL due to autocommit_before_ddl setting
> +ERROR:  internal error: invalid schema kind for getNonTemporarySchemaForCreate: 2
> ```

Looks like #142780 isn't fully fixed.
___
*[`pkg/cmd/roachtest/testdata/pg_regress/jsonb_jsonpath.diffs` line 1394 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/145233#-OP-8dZh5NhlwgEvX7FL:-OP-8dZiAoKYKV9JEGeg:b-8wg9gq) ([raw file](https://github.com/cockroachdb/cockroach/blob/6f780acb813cce431d78b6ee7b331e189b75bad9/pkg/cmd/roachtest/testdata/pg_regress/jsonb_jsonpath.diffs#L1394)):*
> ```
>  ------------------
> - "boolean"
> + "null"
> ```

This doesn't look right. Happy to open an issue if you haven't yet.
___
*[`pkg/cmd/roachtest/testdata/pg_regress/jsonb_jsonpath.diffs` line 1667 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/145233#-OP-ASOtEwnvxKcU1V6e:-OP-ASOtEwnvxKcU1V6f:b-ed2qp3) ([raw file](https://github.com/cockroachdb/cockroach/blob/6f780acb813cce431d78b6ee7b331e189b75bad9/pkg/cmd/roachtest/testdata/pg_regress/jsonb_jsonpath.diffs#L1667)):*
> ```
> + "a\\b"
> + "^a\\b$"
> +(3 rows)
> ```

Here's another diff that probably deserves an issue, along with the case below.
___
*[`pkg/cmd/roachtest/testdata/pg_regress/truncate.diffs` line 508 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/145233#-OP-IuR2ERs-Aaf7hDbZ:-OP-IuR2ERs-Aaf7hDb_:b-ulogef) ([raw file](https://github.com/cockroachdb/cockroach/blob/6f780acb813cce431d78b6ee7b331e189b75bad9/pkg/cmd/roachtest/testdata/pg_regress/truncate.diffs#L508)):*
> ```
>    END
>  $$;
> +ERROR:  relation "NULL" does not exist
> ```

I opened an issue for this here: https://github.com/cockroachdb/cockroach/issues/145411
___
*[`pkg/cmd/roachtest/testdata/pg_regress/rowsecurity.diffs` line 2506 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/145233#-OP-Gq8mCYzrOuVKhkHk:-OP-Gq8mCYzrOuVKhkHl:b-2tvtgy) ([raw file](https://github.com/cockroachdb/cockroach/blob/6f780acb813cce431d78b6ee7b331e189b75bad9/pkg/cmd/roachtest/testdata/pg_regress/rowsecurity.diffs#L2506)):*
> ```
> -(1 row)
> -
> +ERROR:  new row violates row-level security policy for table "document"
> ```

Looks like we weren't supposed to prevent this statement, and similar below.


<!-- Sent from Reviewable.io -->

mgartner reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/145233)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @DrewKimball and @yuzefovich)

___
*[`pkg/cmd/roachtest/testdata/pg_regress/rowsecurity.diffs` line 2506 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/145233#-OP-Gq8mCYzrOuVKhkHk:-OP0nC636Iuq3oIQ0XnQ:bpv2skv) ([raw file](https://github.com/cockroachdb/cockroach/blob/6f780acb813cce431d78b6ee7b331e189b75bad9/pkg/cmd/roachtest/testdata/pg_regress/rowsecurity.diffs#L2506)):*
<details><summary><i>Previously, DrewKimball (Drew Kimball) wrote…</i></summary><blockquote>

Looks like we weren't supposed to prevent this statement, and similar below.
</blockquote></details>

I think this is because the `UPDATE` policy isn't created — it references a different table `category` which we don't support. See the `ERROR:  column "cname" does not exist` diff on line 2486.


<!-- Sent from Reviewable.io -->

yuzefovich reviewed (commented):
Thanks for the careful review!

bors r+

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/145233#-:-OP1J0T5BTCZkpOdrf8j:b-ae1x1u)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @DrewKimball)

___
*[`pkg/cmd/roachtest/testdata/pg_regress/jsonb_jsonpath.diffs` line 1394 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/145233#-OP-8dZh5NhlwgEvX7FL:-OP1Bq4K4c94NLka_qlC:b-eh5awz) ([raw file](https://github.com/cockroachdb/cockroach/blob/6f780acb813cce431d78b6ee7b331e189b75bad9/pkg/cmd/roachtest/testdata/pg_regress/jsonb_jsonpath.diffs#L1394)):*
<details><summary><i>Previously, DrewKimball (Drew Kimball) wrote…</i></summary><blockquote>

This doesn't look right. Happy to open an issue if you haven't yet.
</blockquote></details>

Yep, already filed #145399.
___
*[`pkg/cmd/roachtest/testdata/pg_regress/jsonb_jsonpath.diffs` line 1667 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/145233#-OP-ASOtEwnvxKcU1V6e:-OP1Bup0-K7MSFJFnNP2:bone4fx) ([raw file](https://github.com/cockroachdb/cockroach/blob/6f780acb813cce431d78b6ee7b331e189b75bad9/pkg/cmd/roachtest/testdata/pg_regress/jsonb_jsonpath.diffs#L1667)):*
<details><summary><i>Previously, DrewKimball (Drew Kimball) wrote…</i></summary><blockquote>

Here's another diff that probably deserves an issue, along with the case below.
</blockquote></details>

Filed this one too #145270.
___
*[`pkg/cmd/roachtest/testdata/pg_regress/jsonpath.diffs` line 140 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/145233#-OP-B4IS6cLZAX2aBWb5:-OP1Bybr4Vy71B3c4eDE:brhwx3z) ([raw file](https://github.com/cockroachdb/cockroach/blob/6f780acb813cce431d78b6ee7b331e189b75bad9/pkg/cmd/roachtest/testdata/pg_regress/jsonpath.diffs#L140)):*
<details><summary><i>Previously, DrewKimball (Drew Kimball) wrote…</i></summary><blockquote>

Looks like something strange going on with the normalization, here.
</blockquote></details>

The formatting differences here I deemed as not too important, so I didn't initially file an issue for this. Given you pointed several things out, I filed #145437 which includes representative deviations.
___
*[`pkg/cmd/roachtest/testdata/pg_regress/jsonpath.diffs` line 218 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/145233#-OP-BaTq1dE0Q0j3jCe3:-OP1HUVr9YNJ5DLprm23:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/6f780acb813cce431d78b6ee7b331e189b75bad9/pkg/cmd/roachtest/testdata/pg_regress/jsonpath.diffs#L218)):*
<details><summary><i>Previously, DrewKimball (Drew Kimball) wrote…</i></summary><blockquote>

This is another case where the normalization isn't quite right - the result doesn't round-trip through the jsonpath parser:
```
root@localhost:26257/defaultdb> select '1.type()'::jsonpath;
ERROR: could not parse "1.type()" as type jsonpath: at or near "trailing junk after numeric literal at or near "1.t"": syntax error
SQLSTATE: 42601
DETAIL: source SQL:
1.type()
^
```
</blockquote></details>

Done.
___
*[`pkg/cmd/roachtest/testdata/pg_regress/jsonpath.diffs` line 358 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/145233#-OP-CLqx4ZlXtObbsmlJ:-OP1HYVX2tm-ckrjdktJ:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/6f780acb813cce431d78b6ee7b331e189b75bad9/pkg/cmd/roachtest/testdata/pg_regress/jsonpath.diffs#L358)):*
<details><summary><i>Previously, DrewKimball (Drew Kimball) wrote…</i></summary><blockquote>

This is another one that doesn't round-trip.
</blockquote></details>

Done.
___
*[`pkg/cmd/roachtest/testdata/pg_regress/temp.diffs` line 471 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/145233#-OP-Hqvw6SYZS84rstvu:-OP1IpFo0hHEJ1xitUgM:b-f8g161) ([raw file](https://github.com/cockroachdb/cockroach/blob/6f780acb813cce431d78b6ee7b331e189b75bad9/pkg/cmd/roachtest/testdata/pg_regress/temp.diffs#L471)):*
<details><summary><i>Previously, DrewKimball (Drew Kimball) wrote…</i></summary><blockquote>

Looks like #142780 isn't fully fixed.
</blockquote></details>

Thanks, I meant to file an issue for this one. Done - #145438.


<!-- Sent from Reviewable.io -->
